### PR TITLE
feat(replays): Extend subquery behavior to allow filtering and sorting on static, segment 0 values

### PR DESCRIPTION
**Pull request by cmanallen**

### CODE REVIEW

File: src/sentry/replays/query.py [hunk 1]
```diff
@@ -590,6 +593,32 @@ class ReplayQueryConfig(QueryConfig):
     project = String(query_alias="project_id", is_filterable=False)
 
 
+class ReplaySubqueryConfig(QueryConfig):
```
Comment (JoshFerge) [inline src/sentry/replays/query.py:596]: ah a separate query config is a great way to do this. nice.

**Review by JoshFerge — APPROVED** [review]
makes sense to me. just question about env. nice!!

File: src/sentry/replays/query.py [hunk 2]
```diff
@@ -60,67 +62,49 @@ def query_replays_collection(
...
-            and len(search_filters) == 0
-            and (sort is None or _sort_eligible_for_subquery(sort))
-            and len(conditions) == 0
-            and features.has(
-                "organizations:session-replay-index-subquery", organization, actor=actor
-            )
-        )
-
     conditions = []
     if environment:
```
[the remark this answers is not in the document]
Comment (cmanallen) [reply src/sentry/replays/query.py:71]: I'm on the fence about it.  If you have a strong preference I'm fine mapping it to the 0th segment.

File: src/sentry/replays/query.py (hunk 1 above)
Comment (cmanallen) [reply src/sentry/replays/query.py:596]: 🎉

File: src/sentry/replays/query.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (JoshFerge) [reply src/sentry/replays/query.py:71]: yeah let's follow up & discuss -- i'm cool w/ merging this!

**Pull request merged**